pkg/controller/core.oam.dev/v1alpha2/applicationcontext/applicationcontext_controller.go

```diff
@@ -107,9 +107,10 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	// call into the old ac Reconciler and copy the status back
 	acReconciler := ac.NewReconciler(r.mgr, dm, r.log, ac.WithRecorder(r.record), ac.WithApplyOnceOnlyMode(r.applyMode))
 	reconResult := acReconciler.ACReconcile(ctx, appConfig, r.log)
+
+	appContextPatch := client.MergeFrom(appContext.DeepCopy())
 	appContext.Status = appConfig.Status
-	// always update ac status and set the error
-	err = errors.Wrap(r.client.Status().Update(ctx, appContext), errUpdateAppContextStatus)
+	err = errors.Wrap(r.client.Patch(ctx, appContext, appContextPatch), errUpdateAppContextStatus)
```

Comment:
why it can fix the issue? the status is a sub-resource

Reply:
sorry , i has been modified to be a sub resource  patch，the cause of this error Update() may fail a few times before passing due to the cache being stale,Patch() can avoid